test/mobx.js

```diff
@@ -0,0 +1,364 @@
+import {flattenTree, keyPath} from '../src/util/tree'
+import * as F from 'futil-js'
+import _ from 'lodash/fp'
+import chai from 'chai'
+import sinon from 'sinon'
+import sinonChai from 'sinon-chai'
+import * as lib from '../src'
+import { observable, reaction, toJS, extendObservable } from 'mobx'
+import util from 'util'
+import Promise from 'bluebird'
+const expect = chai.expect
+chai.use(sinonChai)
+
+let traverse = x => x && x.children && x.children.slice() // mobx needs slice
+let treeUtils = F.tree(traverse, keyPath)
+let ContextTreeMobx = (tree, service) => lib.ContextTree(tree, service, undefined, { snapshot: toJS, extend: extendObservable })
+
+describe('mobx', () => {
+  let tree = observable({
```

Comment:
it looks like we're instantiating a couple of trees - we might only need 1 for these tests